src/clients/python/__init__.py

```diff
@@ -1416,53 +1417,62 @@ def _get_results(self, outputs, batch_size, request_id=None):
                     shm_fd = c_int()
                     offset = c_uint64()
                     byte_size = c_uint64()
-                    shm_addr = c_void_p()
+                    shm_addr = c_char_p()
                     shm_key = c_char_p()
                     _raise_if_error(
                         c_void_p(_crequest_get_shared_memory_handle_info(output_format[1], \
                                 byref(shm_addr), byref(shm_key), byref(shm_fd), \
                                 byref(offset), byref(byte_size))))
-                    if (np.prod(shape) * np.dtype(result_dtype).itemsize) < int(byte_size.value/batch_size):
-                        element_byte_size = np.prod(shape) * np.dtype(result_dtype).itemsize
-                    else:
-                        element_byte_size = int(byte_size.value/batch_size)
-                    start_pos = offset.value
-                    if result_dtype != np.object:
-                        cval = shm_addr
-                        for b in range(batch_size):
-                            cval_len = start_pos + element_byte_size
-                            if cval_len == 0:
-                                val = np.empty(shape, dtype=result_dtype)
-                                results[output_name].append(val)
-                            else:
-                                val_buf = cast(cval, POINTER(c_byte * cval_len))[0]
-                                val = np.frombuffer(val_buf, dtype=result_dtype, offset=start_pos)
+                    try:
```

Comment:
Why `try` block here? I don't see there are early returns in the `try` block

Reply:
Or you should move `try` up to cover `_crequest_get_shared_memory_handle_info`. Note that exception may be raised from `_crequest_get_shared_memory_handle_info` after the internal buffer is allocated

Reply:
We allocate a new memory buffer (in the case of GPU shared memory) in _crequest_get_shared_memory_handle_info. If for some reason there is an exception while processing that into a numpy array, we still want this buffer to be destroyed.

Reply:
> Or you should move `try` up to cover `_crequest_get_shared_memory_handle_info`. Note that exception may be raised from `_crequest_get_shared_memory_handle_info` after the internal buffer is allocated

Did you do this?

Reply:
Done